drv/vsc7448/src/lib.rs

```diff
@@ -766,6 +766,98 @@ impl<'a, R: Vsc7448Rw> Vsc7448<'a, R> {
         Ok(())
     }
 
+    /// Implements the VLAN scheme described in RFD 250, with one exception:
+    /// the technician ports are on **every VLAN**, so they can talk to any SP
+    /// without having to go through the CPU port to the Tofino.
+    pub fn configure_vlan_semistrict(&self) -> Result<(), VscError> {
+        const UPLINK: u8 = 49; // DEV10G_0, uplink to the Tofino 2
+
+        const TECHNICIAN_1: u8 = 44;
+        const TECHNICIAN_2: u8 = 45;
+
+        // Enable the VLAN
+        self.write_with(ANA_L3().COMMON().VLAN_CTRL(), |r| r.set_vlan_ena(1))?;
+
+        // By default, there are three VLANs configured in ANA_L3:
+        // 0, 1, and 4095.  We disable all of them, since we only want to
+        // allow very specific VIDs.
+        for vid in [0, 1, 4095] {
+            self.write_port_mask(ANA_L3().VLAN(vid).VLAN_MASK_CFG(), 0)?;
+        }
+
+        // Configure the downstream ports, which each have their own VLANs
+        for p in (0..=52).filter(|&p| p != UPLINK) {
+            let port = ANA_CL().PORT(p);
+
+            // Pick a mask of what ports are conntected to this port
+            let mask = if p == TECHNICIAN_1 || p == TECHNICIAN_2 {
+                // Technician ports are connected to every port!
+                (1 << 53) - 1
+            } else {
+                // SPs are only connected to the Tofino and technician ports
+                (1 << p)
+                    | (1 << UPLINK)
+                    | (1 << TECHNICIAN_1)
+                    | (1 << TECHNICIAN_2)
+            };
+
+            // Configure the 0x1YY VLAN for this port
+            self.write_port_mask(
+                ANA_L3().VLAN(0x100 + p as u16).VLAN_MASK_CFG(),
+                mask,
```

Comment:
Is this function identical to `configure_vlan_strict()` except for this mask? If so, could we split it into a common function that takes a `port -> mask` closure?

Reply:
I was suspicious of this, but did end up liking it (3f0c314ae696b2a8df73b05306682a72f4cb9038)